hathor/wallet/resources/thin_wallet/send_tokens.py

```diff
@@ -173,32 +193,34 @@ def _responseFailed(self, err, tx, request):
             self.manager.stratum_factory.update_jobs()
         else:
             # if we're mining on a thread, stop it
-            request.should_stop_mining_thread = True
+            context.should_stop_mining_thread = True
 
-    def _stratum_deferred_resolve(self, tx: Transaction, request: Request) -> None:
+    def _stratum_deferred_resolve(self, context: _Context) -> None:
         """ Method called after stratum resolves tx proof of work
             We remove the mining data of this tx on stratum and start a new thread to verify the tx
         """
-        funds_hash = tx.get_funds_hash()
-        tx = self.manager.stratum_factory.mined_txs[funds_hash]
+        funds_hash = context.tx.get_funds_hash()
+        context.tx = self.manager.stratum_factory.mined_txs[funds_hash]
         # Delete it to avoid memory leak
         del(self.manager.stratum_factory.mined_txs[funds_hash])
 
-        deferred = threads.deferToThreadPool(reactor, self.manager.pow_thread_pool, self._stratum_thread_verify, tx)
-        deferred.addCallback(self._cb_tx_resolve, request)
-        deferred.addErrback(self._err_tx_resolve, request, 'stratum_resolve')
+        deferred = threads.deferToThreadPool(reactor, self.manager.pow_thread_pool,
+                                             self._stratum_thread_verify, context)
+        deferred.addCallback(self._cb_tx_resolve)
+        deferred.addErrback(self._err_tx_resolve, 'stratum_resolve')
```

Comment:
I'm actually not sure about this one. Because it's an "errback" it will probably not receive the argument returned from the deferred. I'll take a better look.

Reply:
Shouldn't we pass `context` to the errback?

Reply:
I think we should. But I still have to look into the deferred system to better understand how it works.

Reply:
Done.

Reply:
Thanks!